src/core/favorites-state/favorite-state.service.ts

```diff
@@ -1,21 +1,24 @@
 import {Injectable} from '@angular/core';
 import {StorageService} from "../storage/storage.service";
-
 @Injectable({
   providedIn: 'root',
 })
 export class FavoriteStateService {
 
-  readonly FAVORITES_KEY = 'favorites-cities'
-
+  readonly FAVORITES_KEY = 'favorites-cities';
+  readonly MAX_AMOUNT_OF_FAVS = 10;
   constructor(private storage: StorageService) {}
 
   getFavoriteCities(): string[] {
     return this.storage.getItem(this.FAVORITES_KEY);
   }
 
   addFavoriteCity(city: string): void {
-    this.storage.addItem(this.FAVORITES_KEY, city);
+    if(this.isFavoriteCity(city)){
```

Comment:
if(this.isFavoriteCity(city) && this.hasMaxCities()) {
  this.storage.addItem(this.FAVORITES_KEY, city);
}